py3status/modules/weather_owm.py

```diff
@@ -0,0 +1,587 @@
+# -*- coding: utf-8 -*-
+'''
+Ultimately customizable weather module based on the IP-API Geolocation API
+(http://ip-api.com) and the OpenWeatherMap API (https://openweathermap.org).
+
+Requires an API key for OpenWeatherMap (OWM), but the free tier allows you
+enough requests/sec to get accurate weather even up to the second.
+
+Note that this module does require an additional python package,
+pyowm (https://github.com/csparpa/pyowm), and this may be easily
+installed via pip.
+
+This module allows you to specify an icon for nearly every weather scenario
+imaginable. The default configuration options lump many of the icons into
+a few groups, and due to the limitations of UTF-8, this is really as expressive
+as it gets.
+
+I would highly suggest you install an additional font, such as the incredible
+(and free!) Weather Icons font (https://erikflowers.github.io/weather-icons),
+which has icons for most weather scenarios.
+
+For more information, see the documentation
+(ttps://openweathermap.org/weather-conditions) on what weather conditions are
+supported. See the configuration options for how to specify each weather icon.
+
+Configuration parameters:
+    api_key: Your OpenWeatherMap API key.
```

Comment:
I like the required idea but I think we should wait a little.

I'm planning to overhaul the module docstrings.  The basic plan is.

* write new clever parser/checker (including new features like SAMPLE OUTPUT)

* enable docstring rebuilding from parsed data so we can refresh docstrings to pristine states; indentation, section ordering etc

* `py3status modules list` etc to parse docstrings on the fly so it is up-to-date info and we can list all available modules including user custom ones, assuming we pass the include directories

* remove defaults as we can just get them from the module (we may need to add a `DEFAULTS` for 'special' settings in a couple of modules)

* because we have a 'structured' parsed docstring we can also make the modules documentation really nice and pretty

Anyhow that in the future.

We also need to unify config settings and part of this is gatekeeping new ones.  I don't think `api_key` is a problem here because that is what OWM call it.  But it is good to think about